tests/main/presenters/test_search_presenters.py

```diff
@@ -680,3 +681,256 @@ def test_build_lots_and_categories_link_tree_with_lot_and_no_category_filters_ha
+                        )
+                    }
+                ],
+                'selected': True,
+                'service_count': 0,
+                'link': '/g-cloud/search?lot=cloud-software&checkboxTreeExample=option+2'
+            }, {
+                'label': 'Option 2.1',
+                'name': 'checkboxTreeExample',
+                'id': 'checkboxTreeExample-option-2.1',
+                'value': 'option 2.1',
+                'selected': True,
+                'service_count': 0
+            }
+        ]
+
+    @pytest.mark.parametrize(
+        'url', (
+            "/g-cloud/search?lot=cloud-software&checkboxTreeExample=option+2.1",
+            "/g-cloud/search?lot=cloud-software&checkboxTreeExample=option+2.1&parentCategory=option+2"
+        )
+    )
+    def test_build_lots_and_categories_link_tree_with_lot_and_filter_selected(self, url):
+        with self.app.test_request_context(url):
+            tree = build_lots_and_categories_link_tree(
+                self.framework,
+                self.framework['lots'],
+                self.category_filter_group.copy(),
+                flask.request,
+                flask.request.args,
+                g9_builder,
+                'services',
+                'g-cloud-9',
+                Href(flask.url_for('.search_services')),
+                self.search_api_client
+            )
+
+        assert tree == self.tree_with_cloud_software__option_2__option_2_1__selected
+        # Assert, as per the url, that only our specified filter is 'selected' in the output
+        assert(
+            [(child['id'], child['selected']) for child in tree[0]['children'][0]['children'][0]['children']] ==
+            [('checkboxTreeExample-option-2.1', True), ('checkboxTreeExample-option-2.2', False)]
+        )
+
+    def test_build_lots_and_categories_link_tree_with_lot_and_filter_and_same_named_filter_in_tree(self):
+        url = "/g-cloud/search?lot=cloud-software&checkboxTreeExample=option+2.1&parentCategory=option+2"
+
+        self.category_filter_group['filters'] += [
+            {
+                'label': 'Option 7',
+                'name': 'checkboxTreeExample',
+                'id': 'checkboxTreeExample-option-7',
+                'value': 'option 7',
+                'children': [
+                    {
+                        'label': 'Option 2.1',
+                        'name': 'checkboxTreeExample',
+                        'id': 'checkboxTreeExample-option-2.1',
+                        'value': 'option 2.1'
+                    }, {
+                        'label': 'Option 7.1',
+                        'name': 'checkboxTreeExample',
+                        'id': 'checkboxTreeExample-option-7.1',
+                        'value': 'option 7.1'
+                    }
+                ]
+            }
+        ]
+        with self.app.test_request_context(url):
+                tree = build_lots_and_categories_link_tree(
```

Comment:
Curious that flake8 didn't pick up on this extra indenting 🤔

Reply:
Hmmmm, technically valid python I guess, but not something we want, changed 👍